devtools/x/src/installer.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{cargo::Cargo, config::CargoConfig};
+use log::info;
+use std::process::Command;
+
+pub fn install_if_needed_or_die(name: &str, version: &str) {
+    let config = CargoConfig {
+        toolchain: "stable".to_owned(),
+        flags: Option::None,
+    };
+    install_if_needed_or_die_config(&config, name, version);
+}
+
+pub fn install_if_needed_or_die_config(config: &CargoConfig, name: &str, version: &str) {
+    if !check_installed(name, version) {
+        info!("Installing {} {}", name, version);
+        let installres = Cargo::new(config, "install")
+            .arg(name)
+            .arg("--version")
+            .arg(version)
+            .run();
+        if installres.is_err() {
+            info!("Could not install {} {}, check x.toml to ensure tools exist and are not yanked.  exiting.", name, version);
+            std::process::abort();
+        }
+    } else {
+        info!("Not re-installing {} {}", name, version);
+    }
+}
+
+pub fn check_installed(name: &str, version: &str) -> bool {
+    let result = Command::new(name).arg("--version").output();
+    let comp = format!("{} {}", name, version);
+    return result.is_ok()
+        && comp.eq(String::from_utf8_lossy(&result.unwrap().stdout.as_slice()).trim());
```

Comment:
This could probably be simplified a little bit using a `match` or `if let`

Reply:
used match, doesn't feel simpler to be, but probably more canonically rust.